### PR TITLE
[azk-core integration] Including azk-core

### PR DESCRIPTION
**Pull request by saitodisse**
* included azk-core as dependency
* t() helper was moved to azk-core on config class
* config is now a class and configured in azk index

config return this object:

```js
return {
  configInstance: _configAzk,
  config        : _configAzk.getKey.bind(_configAzk),
  get           : _configAzk.getKey.bind(_configAzk),
  set           : _configAzk.setKey.bind(_configAzk),
  t             : _configAzk.t.bind(_configAzk),
};
```

* log is a class too and configured on azk index; It receives config_getKey as a dependency
* utils.net was extracted to azk-core class
* utils.docker is now a class and receives config dependencies
* changed timeout of test "should generate a valid image" to 50s
* `_.rest` was replace by `_.slice` because lodash was updated

### CODE REVIEW

File: src/agent/configure.js [hunk 1]
```diff
@@ -264,8 +264,8 @@ export class Configure extends UIProxy {
 
       // Check ip or generate a new one
       if (use_vm) {
-        var vm_name = config("agent:vm:name");
-        this._interfaces = yield net.getInterfacesIps(vm_name);
+        // var vm_name = config("agent:vm:name");
+        this._interfaces = yield net.getInterfacesIps(/* vm_name */); //FIXME: change params here
```
Comment (nuxlli) [inline src/agent/configure.js]: This point has yet to be implemented right?

File: src/agent/vm.js [hunk 2]
```diff
@@ -524,6 +524,16 @@ var vm = {
...
+
+  _getVMHostonlyInterface(vm_name) { //FIXME: use this here
+    return VM.info(vm_name).then((info) => {
+      if (info.installed) {
+        return info.hostonlyadapter1;
+      }
+      return null;
+    });
+  },
+
```
Comment (nuxlli) [inline src/agent/vm.js]: This probably will be a public method to be used in configure, then it must be without the _.

File: src/config.js [hunk 3]
```diff
@@ -1,192 +1,166 @@
...
+  // Paths to use in config
+  var paths = {
+    logs: path.join(data_path, 'logs'),
+    run : path.join(data_path, 'run'),
+    vm  : path.join(data_path, 'vm'),
+  };
+
+  class Dynamic {
+    constructor(key) { this.key = key; }
+  }
```
Comment (nuxlli) [inline src/config.js]: Dynamic must be imported from `azk-core` and not be declardo here.

File: src/utils/docker.js [hunk 4]
```diff
@@ -1,12 +1,15 @@
...
+    this._path = path;
+    this._config = config;
+  }
+
   // TODO: Add test
-  static resolvePath(target, point = config('agent:vm:mount_point')) {
-    target = Utils.resolve(target);
-    if (config('agent:requires_vm')) {
-      target = path.join(point, target);
+  resolvePath(target, point = this._config('agent:vm:mount_point')) {
```
Comment (nuxlli) [inline src/utils/docker.js]: What is the reason to remove the static?

**Comment by fearenales** [comment]
Can one of the admins verify this patch?

**Comment by fearenales** [comment]
ok to test

**Comment by fearenales** [comment]
ok to test

**Comment by fearenales** [comment]
ok to test

**Comment by fearenales** [comment]
ok to test

File: src/agent/configure.js (hunk 1 above)
Comment (saitodisse) [inline src/agent/configure.js]: yes, to be implemented

**Comment by heitortsergent** [comment]
Although `azk-core` is an important change to help in `azk`'s maintenance in the future, right now we're focusing on improving performance and a few other things, related to issues #379, #382, #385, #386, #387 and #388

**Pull request closed without merging**